core/src/test/java/org/pentaho/platform/engine/core/system/objfac/OSGIObjectFactoryTest.java

```diff
@@ -405,6 +408,44 @@ public void testGetObjectReferencesWithQuery() throws Exception {
     verify( mockContext ).getServiceReferences( String.class, "(name=foo)" );
 
 
+  }
+
+  /*The purpose for the next method is to test the Array Sort Comparator*/
+  @Test
+  public void testGetObjectReference_includingArraySortComparator() throws Exception {
+
+    ServiceReference<String> ref1 = Mockito.mock( ServiceReference.class );
+    ServiceReference<String> ref2 = Mockito.mock( ServiceReference.class );
+    ServiceReference<String> ref3 = Mockito.mock( ServiceReference.class );
+
+    ArrayList<ServiceReference<String>> refs = Stream.of( ref1, ref2, ref3 )
+      .collect( Collectors.toCollection( ArrayList::new ) );
+
+    when( mockContext.getServiceReferences( String.class, null ) ).thenReturn( refs );
+    when( mockContext.getService( ref1 ) ).thenReturn( "ref1" );
+    when( mockContext.getService( ref2 ) ).thenReturn( "ref2" );
+    when( mockContext.getService( ref3 ) ).thenReturn( "ref3" );
+
+    //props - all with different values
+    when( ref1.getProperty( Constants.SERVICE_RANKING ) ).thenReturn( "3" );
```

Comment:
`getProperty( Constants.SERVICE_RANKING )` will return an `Integer` or `null`, not a `String`.
See: https://osgi.org/javadoc/r4v43/core/org/osgi/framework/Constants.html#SERVICE_RANKING

> The value of this property must be of type Integer.